script.js

```diff
@@ -0,0 +1,223 @@
+
+// 6. Americans are odd people: in their buildings, the first floor is actually the first floor and there is no 13th floor (due to superstition).
+// Write a function that given a floor in the american system returns the floor in the european system.
+// With the 1st floor being replaced by the first floor and the 13th floor being removed, the numbers move down to take their place. In case of above 13, they move down by two because there are two omitted numbers below them.
+// Basements (negatives) stay the same as the universal level.
+
+function getTheFloorInTheEuropeanSystem(americanFloorNumber) {
+    if (americanFloorNumber <= 0) {
+        return americanFloorNumber;
+    }
+    if (americanFloorNumber <= 12) {
+        return americanFloorNumber - 1;
+    }
+    if (americanFloorNumber >= 14) {
+        return americanFloorNumber - 2;
+    }
+}
+
+console.log(getTheFloorInTheEuropeanSystem(14));
+
+// 7. In the rock-paper-scissors.js file.
+
+// 8. Write the isValidPassword function.
+// Make sure that the password is at least 8 characters long and contains at least one number and one uppercase letter.
+
+function isLettersNumber(word) {
+    return word.length >= 8;
+
+}
+
+function isUppercaseLetter(word) {
+    const uppercaseLetters =  /[A-Z]/g;
+    const matchingResult = word.match(uppercaseLetters);
+    return matchingResult !== null;
+
+}
+
+function isNumber(word) {
+    const numbers = /\d/g;
+    const matchingResult = word.match(numbers);
+    return matchingResult !== null;
+}
+
+function isValidPassword(password) {
+    return isLettersNumber(password) &&
+        isUppercaseLetter(password) &&
+        isNumber(password);
+}
+
+console.log(isValidPassword('Password123'));
+
+// 9. In the getPasswordStrength.js file.
+
+// 10. Create the getTotalOrderCost function that returns the total cost of the order and accepts the following arguments:
+// - the base cost of the items
+// - whether the customer is a VIP member - they get a 5% discount
+// - how many loyalty points the customer wants to use - each one of them reduces the cost by a cent (the final cost should not be lower than $0)
+// - is the package going to be shipped internationally - shipping internationally costs $10 (otherwise, it costs $5)
+// Make sure to apply the VIP discount before the loyalty points. The VIP discount does not apply to shipping.
+
+function getTheCostForVIPMember(cost) {
+    return cost * 0.05;
+}
+
+function getLoyaltyPointsDiscount(points) {
+    return  points / 100;
+}
+
+function getShippedInternationally(shippingType) {
+    if (shippingType === 'international') {
+        return 10;
+    }
+    return 5;
+}
+
+function getTotalOrderCost(baseCost, isVipMember, loyaltyPoints, shippingType) {
+    const shippingCost = getShippedInternationally(shippingType);
+    let loyaltyPointsDiscount  = getLoyaltyPointsDiscount(loyaltyPoints);
+    if (loyaltyPointsDiscount > baseCost) {
+        loyaltyPointsDiscount = 0;
+    }
```

Comment:
```
if (loyaltyPointsDiscount > baseCost) {
  loyaltyPointsDiscount = 0;
}
```
This is not the way to go. You don't need to remove the `loyaltyPointsDiscount` if it is bigger than the `baseCost`, because at the end you do
```
if (totalCost < 0) {
  return 0;
}
```